sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,934 @@
+      fullPath,
+      this.ruleResourceSerializer
+    );
+
+    return this.buildRuleResponse(response);
+  }
+
+  /**
+   * Lists existing rules.
+   * @param topicName
+   * @param subscriptionName
+   * @param listRequestOptions
+   */
+  async listRules(
+    topicName: string,
+    subscriptionName: string,
+    listRequestOptions?: ListRequestOptions
+  ): Promise<ListRulesResponse> {
+    log.httpAtomXml(
+      `Performing management operation - listRules() with options: ${listRequestOptions}`
+    );
+    const fullPath = this.getSubscriptionPath(topicName, subscriptionName) + "/Rules/";
+    const response: HttpOperationResponse = await this.listResources(
+      fullPath,
+      listRequestOptions,
+      this.ruleResourceSerializer
+    );
+
+    return this.buildListRulesResponse(response);
+  }
+
+  /**
+   * Updates properties on the Rule by the given name based on the given options
+   * @param topicName
+   * @param subscriptionName
+   * @param ruleName
+   * @param ruleOptions Options to configure the Rule being updated.
+   * For example, you can configure the filter to apply on associated Topic/Subscription
+   */
+  async updateRule(
+    topicName: string,
+    subscriptionName: string,
+    ruleName: string,
+    ruleOptions: RuleOptions
+  ): Promise<UpdateRuleResponse> {
+    log.httpAtomXml(
+      `Performing management operation - updateRule() for "${ruleName}" with options: ${ruleOptions}`
+    );
+    const fullPath = this.getRulePath(topicName, subscriptionName, ruleName);
+    const response: HttpOperationResponse = await this.putResource(
+      fullPath,
+      buildRuleOptions(ruleName, ruleOptions),
+      this.ruleResourceSerializer,
+      true
+    );
+
+    return this.buildRuleResponse(response);
+  }
+
+  /**
+   * Deletes a rule.
+   * @param topicName
+   * @param subscriptionName
+   * @param ruleName
+   */
+  async deleteRule(
+    topicName: string,
+    subscriptionName: string,
+    ruleName: string
+  ): Promise<DeleteRuleResponse> {
+    log.httpAtomXml(`Performing management operation - deleteRule() for "${ruleName}"`);
+    const fullPath = this.getRulePath(topicName, subscriptionName, ruleName);
+    const response: HttpOperationResponse = await this.deleteResource(
+      fullPath,
+      this.ruleResourceSerializer
+    );
+
+    return this.buildRuleResponse(response);
+  }
+
+  /**
+   * Creates or updates a resource based on `isUpdate` parameter.
+   * @param name
+   * @param entityFields
+   * @param isUpdate
+   * @param serializer
+   */
+  private async putResource(
+    name: string,
+    entityFields:
+      | InternalQueueOptions
+      | InternalTopicOptions
+      | InternalSubscriptionOptions
+      | InternalRuleOptions,
+    serializer: AtomXmlSerializer,
+    isUpdate: boolean = false
+  ): Promise<HttpOperationResponse> {
+    const webResource: WebResource = new WebResource(this.getUrl(name), "PUT");
+    webResource.body = JSON.stringify(entityFields);
```

Comment:
Looks like this is necessary as the tests are all failing if we skip this, may need a closer look at what's going on - I suspect it's do with types used in interface

Reply:
The tests might be failing because of the type of input being passed needs to change. From what I can see the JSON.stringify here and the JSON.parse inside `executeAtomXmlOperation` can be skipped. The conversion to string and back to json should not be necessary

Reply:
Hm, seems tricky - it's just an intermediary step but is still breaking the feature, can do it later as it seems to be a minor concern

Reply:
Please log an issue for further investigation on this. While we can go with the current approach, it is relevant for us to understand why we need the round trip of JSON stringyfing and parsing.

Reply:
Filed https://github.com/Azure/azure-sdk-for-js/issues/5885